clis/teliod/src/qnap.rs

```diff
@@ -0,0 +1,239 @@
+    println!("{}", response);
+}
+
+fn start_daemon() {
+    if is_teliod_running() {
+        send_response(400, "Application is already running.");
+        return;
+    }
+
+    let teliod_log_file = match fs::OpenOptions::new()
+        .create(true)
+        .write(true)
+        .truncate(true)
+        .open(PathConfig::teliod_log())
+    {
+        Ok(file) => file,
+        Err(_) => {
+            send_response(500, "Failed to open teliod log file.");
+            return;
+        }
+    };
+
+    match Command::new("setsid")
+        .arg(PathConfig::teliod_bin())
+        .arg("daemon")
+        .arg(PathConfig::teliod_cfg())
+        .stdout(Stdio::from(
+            teliod_log_file
+                .try_clone()
+                .expect("Failed to open teliod log file."),
+        ))
+        .stderr(Stdio::from(teliod_log_file))
+        .spawn()
+    {
+        Ok(child) => {
+            let pid = child.id();
+            if fs::write(PathConfig::pid(), pid.to_string()).is_ok() {
+                send_response(200, "Application started successfully.");
+            } else {
+                send_response(500, "Failed to save PID.");
+            }
+        }
+        Err(_) => send_response(500, "Failed to start the application."),
+    }
+}
+
+fn stop_daemon() {
+    if is_teliod_running() {
+        if let Ok(pid) = fs::read_to_string(PathConfig::pid()) {
+            if Command::new("kill").arg(pid.trim()).status().is_ok() {
+                let _ = fs::remove_file(PathConfig::pid());
+                send_response(200, "Application stopped successfully.");
+            } else {
+                send_response(500, "Failed to stop the application.");
+            }
+        }
+    } else {
+        send_response(400, "Application is not running.");
+    }
+}
+
+// TODO: LLT-5712
+fn update_config(_post_data: &str) {
+    // match serde_json::from_str::<serde_json::Value>(post_data)
+    //     .ok()
+    //     .and_then(|json| json.get("config").and_then(|c| c.as_str()))
+    // {
+    //     Some(new_config) => {
+    //         if fs::write(PathConfig::cfg(), new_config).is_ok() {
+    //             send_response(200, "Configuration updated successfully.");
+    //         } else {
+    //             send_response(500, "Failed to update configuration.");
+    //         }
+    //     }
+    //     None => send_response(400, "Invalid configuration data."),
+    // }
+}
+
+fn get_status() {
+    if is_teliod_running() {
+        match Command::new(PathConfig::teliod_bin())
+            .arg("get-status")
+            .output()
+        {
+            Ok(output) => {
+                let status = String::from_utf8_lossy(&output.stdout).to_string();
+                send_response(200, json!({ "status-report": status }));
+            }
+            Err(_) => send_response(500, "Failed to retrieve status."),
+        }
+    } else {
+        send_response(400, "Application is not running.");
+    }
```

Comment:
Maybe it would be better to directly send it to the socket?

Reply:
? I didn't understand, can you explain?

Reply:
Soo, currently this methods is teliod running teliod which sends message to another teliod through the socket. I guess you could send this message directly instead.

Reply:
On the other had `cgi` would still have to call the teliod binary to start the daemon.

Reply:
Reused daemon socket where aplicable. As @ghztomash mentioned for daemon start we still have to run `teliod` as a command.

Reply:
qweqwe

Reply:
asdasd